### DISCUSSION TITLE
DB2 for i results view enhancements

### DISCUSSION THREAD

**Discussion by WBSpmichal** [codefori/vscode-ibmi#2335] [Ideas]
currently when running a Db2 query, the results panel is missing some useful functionality I have seen in other query programs. 

**Display as table**
It would be nice if the result set were displayed as a proper table, which would make it easier to copy subsets of cells for pasting into excel
It would then also be nice to be able to resize columns and rows. 

**Column Filtering** 
It would be useful if each column allowed for further filtering within a result set, kind of like filtering columns in excel

**Column Sort by** 
Same as above, but for column sort by

**Alternating row highlight**
it would be nice if alternating rows were displayed with a slightly different background color to help with readability. 

**Total record count**
it currently shows the number of loaded rows, but it would be nice if it also showed the total row count, maybe like "Loaded 100 of 5000 records" or something. 

**Save to File button**
I am aware of the --cvs functionality, but it would be nice if there was something on the UI to hint at that functionality, even if its just a button that adds the --cvs line and re-runs the query.

**Comment by rogersonra** [comment]
Another enhancement would be a row id in the first column to know which row within the resultset the cursor is on.  Please point it out if this exists and I just haven't found it.

Thanks,

Rob

**Comment by WBSpmichal** [comment]
I will also add to this request the ability to display the column title and column text at the same time, like it does with the iaccess sql tool. 

![image](https://github.com/user-attachments/assets/bdcef252-c569-4b1d-8020-97af6d822894)
